c10/core/impl/TorchDispatchModeTLS.h

```diff
@@ -6,20 +6,57 @@
 namespace c10 {
 namespace impl {
 
+enum class TorchDispatchModeKey : int8_t {
+  PROXY = 0,
+  FAKE = 1,
+  NUM_MODE_KEYS = 2
+};
+
 struct C10_API TorchDispatchModeTLS {
+  // This API is NOT invariant safe.
+  // If you're pushing an infra mode onto the stack, we expect
+  // you to use set_mode
   static void push_onto_stack(std::shared_ptr<SafePyObject> mode);
```

Comment:
It seems like it would be safer to rename this method, to force you to audit all the call sites. This might help you diagnose your test failures.

Reply:
I believe I have tests passing (minus a public python bindings test due to me accidentally making `TorchDispatchModeKey` public, and what I believe is a flaky inductor test I need to re-run).

But I can rename this function to something clearer, like `push_non_infra_mode_onto_stack`